presto-testing/src/main/java/io/prestosql/testing/AbstractTestAggregations.java

```diff
@@ -89,6 +89,32 @@ public void testAggregationPushdownThroughOuterJoinNotFiringInCorrelatedAggregat
                 "VALUES 1");
     }
 
+    /**
+     * This case tests that Aggregation isn't incorrectly pushed down into inner source of JoinNode
+     * in the case when it uses symbols from the outer source of JoinNode.
+     */
+    @Test
+    public void testAggregationUsingOuterTableSymbols()
+    {
+        assertQuery(
+                "SELECT max(r.regionkey * n.nationkey) FROM (SELECT DISTINCT regionkey FROM region) r LEFT JOIN nation n ON n.regionkey = r.regionkey  GROUP BY r.regionkey",
+                "VALUES 0, 24, 42, 69, 80");
+    }
+
+    /**
+     * This case tests that Aggregation count(*) can be successfully pushed down into inner source of JoinNode.
```

Comment:
This test creates necessary circumstances for the `count(*)` to be pushed down, but doesn't test it's being pushed. I think some explanation is useful, so do not remove. Reword.